web-app/js/portal/filter/BaseFilterPanel.js

```diff
@@ -101,28 +101,51 @@ Portal.filter.BaseFilterPanel = Ext.extend(Ext.Panel, {
 
 Portal.filter.BaseFilterPanel.newFilterPanelFor = function(cfg) {
 
-    var newFilterPanel;
+    var type = cfg.filter.getType().toLowerCase();
 
-    if (cfg.filter.getType() === "String") {
-        newFilterPanel = new Portal.filter.ComboFilterPanel(cfg);
+    // Todo - DN: Remove when loading filters from DBis removed
+    // Portal types --> GeoServer types (temporary)
+    if (type == "daterange") {
+        type = 'datetime'
     }
-    else if (cfg.filter.getType() == "Date") {
-        newFilterPanel = new Portal.filter.DateFilterPanel(cfg);
+    else if (type === "boundingbox") {
+        type = 'geometrypropertytype'
     }
-    else if (cfg.filter.getType() == "DateRange") {
-        newFilterPanel = new Portal.filter.DateFilterPanel(cfg);
+    else if (type === "number") {
```

Comment:
Could be I'm being paranoid, but what do you think about using the == operator instead of the === operator? That way we get around any potential type comparison problems.